smart-contracts/osmosis/contracts/cl-vault/src/helpers/getters.rs

```diff
@@ -38,26 +37,56 @@ pub fn get_asset0_value(
         .spot_price
         .parse()?;
 
-    let total = token0
-        .checked_add(token1.multiply_ratio(spot_price.denominator(), spot_price.numerator()))?;
+    let total = tokens_provided.0.checked_add(
+        tokens_provided
+            .1
+            .multiply_ratio(spot_price.denominator(), spot_price.numerator()),
+    )?;
 
     Ok(total)
 }
 
-pub fn get_twap_price(
+pub fn get_position_balance(
     storage: &dyn Storage,
     querier: &QuerierWrapper,
-    env: &Env,
+) -> Result<(f64, f64), ContractError> {
+    let position = get_position(storage, querier)?;
+    let asset0_amount = Uint128::from_str(&position.clone().asset0.unwrap_or_default().amount)?;
+    let asset1_amount = Uint128::from_str(&position.clone().asset1.unwrap_or_default().amount)?;
+
+    // Handle cases where either asset amount is zero
+    if asset0_amount.is_zero() && asset1_amount.is_zero() {
```

Comment:
This is probably unnecessary as we are not making checked operations

Reply:
:D this is the only necessary check. If both assets are 0, then asset_0_value becomes 0 and we will probably trigger a panic due to division by zero. So the check for both assets is needed.
The following 2 checks are not needed as the computation below handles these cases

Reply:
and we should use checked operations probably

Reply:
addressed, now we return Decimals here, which obbligates the invoker of this getter (working in Uint128) to then parse it. Its a bit worse there, but still better from a func perspective imo. Let me know if looks good to you